// list/model/listModel.js
var util = require("util");
var model = require("model");



function listModel(req,res)
{
    this.req = req;
    this.res = res;
    var that = this;
       
    this.index = function()
    {
        that.result.name = "testname";
        that.db.view("shoppinglist/allList",function(err,res)
        {
            var table = new(that.table);
            var data = addToData(res); 
            var tableResult = table.create(
            {
                "head"      : ["Anzahl","Name"],
                "option"    : {"edit":true,"create":true},
                "path"      :  "/admin/list/",
                "data"      :  data.list
            });
            that.result.list = tableResult; 
            that.exports(that.result);  
          
        });
        
    };
    
    this.create = function()
    {
        that.result.name="test im Create";
        that.db.save({
                    "type":"list",
                    "content":[
                        {
                            "name":"Eier",
                            "amount": "400",
                            "unity":"stk",
                            "c_price":"550"
                        },
                        {
                            "name":"Saure Sahne",
                            "amount": "250g",
                            "unity":"g",
                            "c_price":"60"
                        }
                    ]
                    },
                function(err,res){
                    if(!err)
                    {
                        that.result.name = "passt ja";
                        util.debug("wurde eingetragen");
                    }
                    else
                    {
                        util.debug(err);
                    }
                });
        that.exports(that.result);
    };
    this.edit = function()
    {
        return that.result;
    }
    
    var addToData = function(res)
    {
        that.result.list = [];
        for(i in res)
            {
                
                that.result.list.push(
                {
                    "_id" : res[i].id,
                    "row" : [res[i].value.content.length,"Liste "+i]        
                 });
                
            }
            return that.result;
    }

}
listModel.prototype = new model();
 //util.inherits(listModel,model);

module.exports = listModel;
