// list/model/listModel.js
var jade = require("jade");
var cradle = require("mycradle");
var util = require("util");
var event = require("events").EventEmitter;


function listModel(req,res)
{
    var db = new cradle();

    var result = 
            {
                "form":[],
                "list":[]
            };    
    this.req = req;
    this.res = res;
    var that = this;
       
    this.index = function()
    {
        result.name = "testname";
        db.view("shoppinglist/allList",function(err,res)
        {
          sendBack(addToData(res));  
          
        });
        
    };
    
    this.create = function()
    {
        result.name="test im Create";
        db.save({
                    "type":"list",
                    "content":[
                        {
                            "name":"Eier",
                            "amount": "400",
                            "unity":"stk",
                            "c_price":"550"
                        },
                        {
                            "name":"Saure Sahne",
                            "amount": "250g",
                            "unity":"g",
                            "c_price":"60"
                        }
                    ]
                    },
                function(err,res){
                    if(!err)
                        {
                            result.name = "passt ja";
                            util.debug("wurde eingetragen");
                        }
                        else
                            {
                                util.debug(err);
                            }
                });
        return result;
    };
    this.edit = function()
    {
        return result;
    }
    
    var addToData = function(res)
    {
        util.debug("Liste: "+result.list);
        for(i in res)
            {
                var tmpArr = [res[i].value.content.length,"Liste "+i];
                result.list.push(tmpArr);
            }
            return result;
    }
    
    var sendBack = function(data)
    {
        that.emit("module_ready",data); 
    }
}
util.inherits(listModel,event); 



module.exports = listModel;
